### PR TITLE
Update `rstest` syntax

### PR DESCRIPTION
**Pull request by jobarr-amzn**
*Issue #, if available:* #155

*Description of changes:* I was looking at old issues here and bumped into #155- but I saw only three examples of the old `rstest`. Might as well clean them up?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jobarr-amzn** [comment]
We should not merge this until after https://github.com/amazon-ion/ion-rust/pull/479 if still relevant after that.

File: src/value/owned.rs
Comment (jobarr-amzn) [inline src/value/owned.rs:806]: Compare to before the builder macros:
https://github.com/amazon-ion/ion-rust/blob/9019a2eaf972ad6e24c647fd54af00a1b022682b/src/value/owned.rs#L739-L764

The builder macros make this so nice!

**Review by almann — APPROVED** [review]
🚢 LGTM, good to see us modernizing this stuff.

**Pull request merged**
